multibody/kinematic/kinematic_evaluator.h

```diff
@@ -0,0 +1,93 @@
+#pragma once
+
+#include "drake/multibody/plant/multibody_plant.h"
+#include "drake/systems/framework/context.h"
+
+namespace dairlib {
+namespace multibody {
+
+/// Virtual class to represent arbitrary kinematic evaluations
+/// Evaluations are defined by some function phi(q). Implementations
+/// must generate phi(q), J(q) (the Jacboian w.r.t. velocities v), 
+/// d/dt phi(q), and [d/dt J] * v
```

Comment:
Probably a typo here. `d/dt phi(q)` seems to be implemented in `EvalFullTimeDerivative`, so the derived classes don't need to implement it